### PR TITLE
Channel in config

### PR DESCRIPTION
**Pull request by khalford**
### Description:

Changed scheduled reminders to post in a channel taken from the config. This allows you to run the app in different workspaces. Channel IDs are workspace specific so the hardcoded channel would make the app error in other workspaces.

Other changes:
Renamed "application" to "app" for ease of reading.
Merged custom exceptions into a single error as they are all to do with the config.

---

### Submitter:

Have you done the following?:

* [x] Labeled the pull request from the following? `major | minor | patch` or `documentation | workflow`
* [x] Updated the documentation and template config files?
* [x] Added unit tests for new / untested code?
* [x] Verified the relevant staging Docker image runs on `dev-cloud-chatops.nubes.rl.ac.uk`?

New / Existing features:
* [ ] Written integration tests?
* [ ] Configured the Slack Application with new commands or permission scope changes?
* [x] Updated the `version.txt` and `docker-compose.yml` files?

---

### Reviewer:

Have you checked the following?:
* [ ] Version change is appropriate to the code changes? Semantic versioning documentation [here](https://semver.org/)
* [ ] Unit test code coverage is acceptable?
* [ ] Do the CI jobs pass?

### CODE REVIEW

**Comment by khalford** [comment]
CodeCov failure can be ignored as the tests will be included in a future PR

File: cloud-chatops/tests/test_read_data.py [hunk 1]
```diff
@@ -99,5 +111,15 @@ def test_validate_required_files_fail_users(mock_get_token, mock_get_config):
...
-    with pytest.raises(UserMapNotGiven):
+    with pytest.raises(ErrorInConfig):
+        validate_required_files()
+
+
+@patch("read_data.get_config")
+@patch("read_data.get_token")
+def test_validate_required_files_fail_channel(mock_get_token, mock_get_config):
+    """Test the validate files function"""
+    mock_get_token.side_effect = ["mock_bot", "mock_app", "mock_github"]
```
Comment (DavidFair) [inline cloud-chatops/tests/test_read_data.py:122]: I think the warning from codecov is valid, I can't see us access `channel` in any tests to check an error is thrown?

**Review by DavidFair — CHANGES_REQUESTED** [review]

**Review by meoflynn — CHANGES_REQUESTED** [review]

File: cloud-chatops/tests/test_read_data.py [hunk 2]
```diff
@@ -79,7 +91,7 @@ def test_validate_required_files_fail_repo(mock_get_token, mock_get_config):
     """Test the validate files function"""
     mock_get_token.side_effect = ["mock_bot", "mock_app", "mock_github"]
     mock_get_config.side_effect = [{}, {"github1": "slack1"}]
-    with pytest.raises(RepositoriesNotGiven):
+    with pytest.raises(ErrorInConfig):
         validate_required_files()
```
[the remark this answers is not in the document]
Comment (khalford) [reply cloud-chatops/tests/test_read_data.py:96]: The test is there but not set up properly, I have fixed it

File: cloud-chatops/tests/test_read_data.py (hunk 1 above)
Comment (khalford) [reply cloud-chatops/tests/test_read_data.py:122]: Same as Megans comment. The test is there but not set up right. I have fixed this

**Review by DavidFair — APPROVED** [review]

**Review by meoflynn — APPROVED** [review]

**Pull request merged**